cmd/status/printers/table_printer.go

```diff
@@ -0,0 +1,385 @@
+			name:   "namespace",
+			header: "NAMESPACE",
+			width:  10,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
+				namespace := resource.Identifier.Namespace
+				if len(namespace) > availableWidth {
+					namespace = namespace[:availableWidth]
+				}
+				_, err := fmt.Fprint(w, namespace)
+				return len(namespace), err
+			},
+		},
+		{
+			// Column containing the resource type and name. Currently it does not print group or version
+			// since those are rarely needed to uniquely distinguish two resources from each other. Just name
+			// and kind should be enough in almost all cases and saves space in the output.
+			name:   "resource",
+			header: "RESOURCE",
+			width:  40,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
+				text := fmt.Sprintf("%s/%s", resource.Identifier.GroupKind.Kind, resource.Identifier.Name)
+				if len(text) > availableWidth {
+					text = text[:availableWidth]
+				}
+				_, err := fmt.Fprint(w, text)
+				return len(text), err
+			},
+		},
+		{
+			// Column containing the status of the resource as computed by the status and polling
+			// libraries.
+			name:   "status",
+			header: "STATUS",
+			width:  10,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
+				s := resource.Status.String()
+				if len(s) > availableWidth {
+					s = s[:availableWidth]
+				}
+				color, setColor := colorForTableStatus(resource.Status)
+				var outputStatus string
+				if setColor {
+					outputStatus = sPrintWithColor(color, s)
+				} else {
+					outputStatus = s
+				}
+				_, err := fmt.Fprint(w, outputStatus)
+				return len(s), err
+			},
+		},
+		{
+			// Column containing the conditions available on the resource. It uses colors to
+			// show the status of each of the conditions.
+			name:   "conditions",
+			header: "CONDITIONS",
+			width:  40,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
+				u := resource.Resource
+				if u == nil {
+					return fmt.Fprintf(w, "-")
+				}
+
+				conditions, found, err := unstructured.NestedSlice(u.Object, "status", "conditions")
+				if !found || err != nil || len(conditions) == 0 {
+					return fmt.Fprintf(w, "<None>")
+				}
+
+				realLength := 0
+				for i, cond := range conditions {
+					condition := cond.(map[string]interface{})
+					conditionType := condition["type"].(string)
+					conditionStatus := condition["status"].(string)
+					var color color
+					switch conditionStatus {
+					case "True":
+						color = GREEN
+					case "False":
+						color = RED
+					default:
+						color = YELLOW
+					}
+					remainingWidth := availableWidth - realLength
+					if len(conditionType) > remainingWidth {
+						conditionType = conditionType[:remainingWidth]
+					}
+					_, err := fmt.Fprint(w, sPrintWithColor(color, conditionType))
```

Comment:
no if setcolor here?

Reply:
In this case we know that we will always print with colors.